pkg/rdt/rdt.go

```diff
@@ -284,27 +308,39 @@ func (c *control) getClassTasks(name string) ([]string, error) {
 	return []string{}, nil
 }
 
-func (c *control) readRdtFile(rdtPath string) ([]byte, error) {
-	return ioutil.ReadFile(filepath.Join(c.info.resctrlPath, rdtPath))
-}
+func (r *resctrlGroup) AddPids(pids ...string) error {
+	f, err := os.OpenFile(r.path("tasks"), os.O_WRONLY, 0644)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
 
-func (c *control) writeRdtFile(rdtPath string, data []byte) error {
-	if err := ioutil.WriteFile(filepath.Join(c.info.resctrlPath, rdtPath), data, 0644); err != nil {
-		return c.cmdError(err)
+	for _, pid := range pids {
+		if _, err := f.WriteString(pid + "\n"); err != nil {
+			unwrapped := err
+			if pathError, ok := err.(*os.PathError); ok {
+				unwrapped = pathError.Unwrap()
+			}
+			if unwrapped == syscall.ESRCH {
+				log.Debug("no task %s", pid)
+			} else {
+				return rdtError("failed to assign processes %v to class %q: %v", pids, r.name, rdt.cmdError(err))
+			}
```

Comment:
Isn't the standard `errors` package meant to simplify this kind of error checks, with something like
```
    if errors.Is(err, syscall.ESRCH) {
        log.Debug(...)
    } else {
        return rdtError(...)
    }
```

Reply:
Thanks for pointing this out. I just blindly copy-pasted the old badly-written code without thinking. Will fix this...

Reply:
Fixed